agent/rpcclient/health/health.go

```diff
@@ -81,8 +81,16 @@ func (c *Client) Notify(
 	ch chan<- cache.UpdateEvent,
 ) error {
 	cacheName := c.CacheName
-	if req.Ingress {
-		cacheName = c.CacheNameIngress
+	if req.Ingress || req.Source.Node != "" {
```

Comment:
It's not a straight replacement, but it would be neat if you could share this potentially variable conditional line with the same line in `UseStreaming` to avoid drift should either one pick up another clause.